pkg/cloudprovider/types.go

```diff
@@ -80,14 +80,29 @@ type InstanceType interface {
 	// Overhead is the amount of resource overhead expected to be used by kubelet and any other system daemons outside
 	// of Kubernetes.
 	Overhead() v1.ResourceList
-	// Price is a metric that is used to optimize pod placement onto nodes as well as determining if replacing a node with
-	// a cheaper instance type is possible when performing consolidation.
-	Price() float64
+	// Price is a metric that is used to optimize pod placement onto nodes.  This can be an actual monetary price per hour
+	// for the instance type, or just a weighting where lower 'prices' are preferred.
+	Price(filters ...func(Offering) bool) float64
 }
 
 // An Offering describes where an InstanceType is available to be used, with the expectation that its properties
 // may be tightly coupled (e.g. the availability of an instance type in some zone is scoped to a capacity type)
-type Offering struct {
-	CapacityType string
-	Zone         string
+type Offering interface {
```

Comment:
I don't think this needs to be an interface if the price parsing is shifted out of the cloud provider.